azure/service.ts

```diff
@@ -21,23 +21,241 @@ import { RunError } from "@pulumi/pulumi/errors";
 import * as shared from "./shared";
 
 import * as docker from "@pulumi/docker";
-import * as config from "./config";
+import * as utils from "./utils";
 
 export class Service extends pulumi.ComponentResource implements cloud.Service {
     public readonly name: string;
+    public readonly groups: azure.containerservice.Group[];
 
     public readonly endpoints: pulumi.Output<cloud.Endpoints>;
     public readonly defaultEndpoint: pulumi.Output<cloud.Endpoint>;
 
     public readonly getEndpoint: (containerName?: string, containerPort?: number) => Promise<cloud.Endpoint>;
 
     constructor(name: string, args: cloud.ServiceArguments, opts?: pulumi.ResourceOptions) {
-        super("cloud:service:Service", name, {}, opts);
+        let containers: cloud.Containers;
+        if (args.image || args.build || args.function) {
+            if (args.containers) {
+                throw new Error(
+                    "Exactly one of image, build, function, or containers must be used, not multiple");
+            }
+            containers = { "default": args };
+        }
+        else if (args.containers) {
+            containers = args.containers;
+        }
+        else {
+            throw new Error(
+                "Missing one of image, build, function, or containers, specifying this service's containers");
+        }
 
-        this.getEndpoint = _ => { throw new Error("Method not implemented."); };
+        const replicas = args.replicas === undefined ? 1 : args.replicas;
+        if (replicas !== 1) {
+            throw new RunError("Only a single replicable is supposrted in Azure currently.")
+        }
 
-        throw new Error("Method not implemented.");
+        super("cloud:service:Service", name, {
+            containers: containers,
+            replicas: replicas,
+        }, opts);
+
+        const { groups, endpoints, defaultEndpoint } = createGroups(
+            this, name, args.host, containers);
+
+        this.groups = groups;
+        this.endpoints = endpoints;
+        this.defaultEndpoint = defaultEndpoint;
+
+        this.getEndpoint = async (containerName, containerPort) => {
+            return getEndpointHelper(endpoints.get(), containerName, containerPort);
+        };
+    }
```

Comment:
Don't we need to call `this.registerResourceOutputs` at the end here? Otherwise the UI will show this thing as `creating` until the plan ends.